scripts/host_cap.py

```diff
@@ -440,6 +467,122 @@ def add_constituent_vars(cap, host_model, suite_list, run_env):
+    >>> tendency_vars = [Var({'local_name':'qv_tend', 'standard_name':'tendency_of_water_vapor_mixing_ratio', 'units':'kg kg-1 s-1', 'dimensions':'(horizontal_dimension, vertical_layer_dimension)', 'type':'real', 'kind':'kind_phys'}, _API_SOURCE, _API_DUMMY_RUN_ENV)]
+    >>> tendency_vars.append(Var({'local_name':'qc_tend', 'standard_name':'tendency_of_cloud_liquid', 'units':'kg kg-1 s-1', 'dimensions':'(horizontal_dimension, vertical_layer_dimension)', 'type':'real', 'kind':'kind_phys'}, _API_SOURCE, _API_DUMMY_RUN_ENV))
+    >>> tendency_vars.append(Var({'local_name':'qc_tend', 'standard_name':'typotendency_of_cloud_liquid', 'units':'kg kg-1 s-1', 'dimensions':'(horizontal_dimension, vertical_layer_dimension)', 'type':'real', 'kind':'kind_phys'}, _API_SOURCE, _API_DUMMY_RUN_ENV))
+    >>> check_tendency_variables(tendency_vars, const_vars)
+    ("\\nInvalid tendency variable 'tendency_of_water_vapor_mixing_ratio'. All constituent tendency variable standard names must be of the form 'tendency_of_<constituent_stdname>'\\nInvalid tendency variable 'tendency_of_cloud_liquid'. All constituent tendency variable standard names must be of the form 'tendency_of_<constituent_stdname>'\\nInvalid tendency variable 'typotendency_of_cloud_liquid'. All constituent tendency variable standard names must be of the form 'tendency_of_<constituent_stdname>'", 1)
+    """
+    errflg = 0
+    errmsg = ''
+
+    for tend_var in tend_vars:
+        prop_error = False
+        valid_var = False
+        tend_stdname = tend_var.get_prop_value('standard_name')
+        tend_units = tend_var.get_prop_value('units')
+        if not tend_stdname.startswith('tendency_of_'):
+            # Tendency standard name is invalid
+            errmsg += f"\nInvalid tendency variable '{tend_stdname}'."
+            errmsg += " All constituent tendency variable standard names must "
+            errmsg += "be of the form 'tendency_of_<constituent_stdname>'"
+            errflg = 1
+            continue
+        tend_stdname_split = tend_stdname.split('tendency_of_')[1]
+        tend_dimensions = tend_var.get_dimensions()
+        tend_type = tend_var.get_prop_value('type')
+        tend_kind = tend_var.get_prop_value('kind')
+        for const_var in const_vars:
+            const_stdname = const_var.get_prop_value('standard_name')
+            const_units = const_var.get_prop_value('units')
+            const_dimensions = const_var.get_dimensions()
+            const_type = const_var.get_prop_value('type')
+            const_kind = const_var.get_prop_value('kind')
+            if tend_stdname_split == const_stdname:
+                # We found a match! Check units
+                if tend_units.split('s-1')[0].strip() != const_units:
```

Comment:
This assumes that `s-1` is always at the end. But it is entirely valid (from a math standpoint) to write `kg s-1 m-2` instead of `kg m-2 s-1`. And it's of course also valid to write `m s-2` for a tendency of the constituent had `m s-1` ...

Probably not an issue, but we should document this in the updated (!) CCPP technical documentation